### ISSUE TITLE
[QUESTION] How to build AOSP TrebleDroid with Gapps

### ISSUE THREAD

**Issue by SowTag** [captaincookie200/treble_yaap#2]
Hey there, ahnet. I know this is not the place to ask but I was wondering if you could teach me how to build a stock AOSP TrebleDroid GSI with Google Apps.

Last time I tried I cloned TrebleDroid/treble_experimentations and installed all required packages.

I modified the build.sh script to execute `buildVariant treble_arm64_bgN-userdebug` and was expecting an A/B build without superuser and with Google Apps. I got a vanilla A/B build with superuser.

Reading your README I found the TARGET_BUILD_GAPPS env variable, I'm now doing another build with the instructions in your README while I write this. Whether or not this works I'll come back here.

TLDR: please teach me how to build a stock aosp trebledroid gsi with google apps because I can't get it to work.

Thanks!

**Comment by realahnet** [comment]
> Hey there, ahnet. I know this is not the place to ask but I was wondering if you could teach me how to build a stock AOSP TrebleDroid GSI with Google Apps.
> 
> Last time I tried I cloned TrebleDroid/treble_experimentations and installed all required packages.
> 
> I modified the build.sh script to execute `buildVariant treble_arm64_bgN-userdebug` and was expecting an A/B build without superuser and with Google Apps. I got a vanilla A/B build with superuser.
> 
> Reading your README I found the TARGET_BUILD_GAPPS env variable, I'm now doing another build with the instructions in your README while I write this. Whether or not this works I'll come back here.
> 
> TLDR: please teach me how to build a stock aosp trebledroid gsi with google apps because I can't get it to work.
> 
> Thanks!

not sure if TARGET_BUILD_GAPPS would work for aosp. The thing your lacking is gapps itself. You can clone the gapps that nazim uses for his crdroid builds by cloning the repo.

`git clone https://gitlab.com/arrowos-project/android_vendor_gapps.git vendor/gapps`

and give it a go again.

As for removing Su you can use [this patch](https://github.com/ahnet-69/treble_yaap/blob/13.0/patches/personal/platform_build/0001-remove-su-from-no-su-builds.patch). Make sure you apply this patch in build/target.

I hoped it helped

**Comment by realahnet** [comment]
> > Hey there, ahnet. I know this is not the place to ask but I was wondering if you could teach me how to build a stock AOSP TrebleDroid GSI with Google Apps.
> > Last time I tried I cloned TrebleDroid/treble_experimentations and installed all required packages.
> > I modified the build.sh script to execute `buildVariant treble_arm64_bgN-userdebug` and was expecting an A/B build without superuser and with Google Apps. I got a vanilla A/B build with superuser.
> > Reading your README I found the TARGET_BUILD_GAPPS env variable, I'm now doing another build with the instructions in your README while I write this. Whether or not this works I'll come back here.
> > TLDR: please teach me how to build a stock aosp trebledroid gsi with google apps because I can't get it to work.
> > Thanks!
> 
> not sure if TARGET_BUILD_GAPPS would work for aosp. The thing your lacking is gapps itself. You can clone the gapps that nazim uses for his crdroid builds by cloning the repo.
> 
> `git clone https://gitlab.com/arrowos-project/android_vendor_gapps.git vendor/gapps`
> 
> and give it a go again.
> 
> As for removing Su you can use [this patch](https://github.com/ahnet-69/treble_yaap/blob/13.0/patches/personal/platform_build/0001-remove-su-from-no-su-builds.patch). Make sure you apply this patch in build/target.
> 
> I hoped it helped

Make sure to adapt the no su patch for aosp it would fail otherwise

**Comment by SowTag** [comment]
Thanks. I manually removed su from PRODUCT_PACKAGES_DEBUG instead of applying your patch because the structure of my .mk files was different.

I cloned ArrowOS' Gapps and I'm now trying another build.

Sadly I don't know whether I can exclude packages from there because it will probably add quite a lot of them increasing image size.

Will reach out in a few hours, thank you!

**Comment by SowTag** [comment]
Update: Build went well but I still got it without Google apps. I cloned the ArrowOS repo but still nothing. I'll now try building with the TARGET_BUILD_GAPPS env, I don't know what else to do otherwise.

**Comment by realahnet** [comment]
> Update: Build went well but I still got it without Google apps. I cloned the ArrowOS repo but still nothing. I'll now try building with the TARGET_BUILD_GAPPS env, I don't know what else to do otherwise.

alr remember how you cloned gapps to vendor/gapps.

delete vendor/gapps

and try vendor/gms

if it still doesnt work you can join telegram group where phh or someone more knowledgeable could help you

**Comment by realahnet** [comment]
I forgot this existed. The problem was fixed so closing issue now.
